### PR TITLE
fix: Corrects CSS output being overwritten when .cjs ext used

### LINKED ISSUE #883: .cjs extension in `main` package.json field causes imported CSS file output to be overwritten
Using a `.cjs` extension in the `main` field of `package.json` (e.g. `"main": "./dist/foo.cjs"`) as per README causes microbundle to swallow any .css or .scss files imported

## Reproduction steps

Repo: https://github.com/lokimckay-references/microbundle-883

1. `npm init -y`
1. `npm install microbundle --save-dev`
1. add `styles.css` and `index.js` files
1. `import "./styles.css"` within `index.js`
1. copy source/exports/main/module/unpkg fields into `package.json` as per README
    ```
    {
      "name": "foo",                     // your package name
      "type": "module",
      "source": "src/foo.js",            // your source code
      "exports": "./dist/foo.modern.js", // where to generate the modern bundle (see below)
      "main": "./dist/foo.cjs",          // where to generate the CommonJS bundle
      "module": "./dist/foo.module.js",  // where to generate the ESM bundle
      "unpkg": "./dist/foo.umd.js",      // where to generate the UMD bundle (also aliased as "umd:main")
      "scripts": {
        "build": "microbundle",          // compiles "source" to "main"/"module"/"unpkg"
        "dev": "microbundle watch"       // re-build when source files change
      }
    }
    ```
1. update `source` field to point to `index.js`
1. `npm run build`

## Actual output

- All compiled formats of index.js
- Below error message

```
The emitted file "foo.cjs" overwrites a previously emitted file of the same name.
```

image

## Expected output

- All compiled formats of index.js
- **styles.css**
- No error message

## Workaround

Change the file extension of the `main` field from `./dist/foo.cjs` -> `./dist/foo.js`
= achieves expected output

### PR DESCRIPTION
**Pull request by rschristian**
Fixes #883

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -483,7 +481,7 @@ function createConfig(options, entry, format, writeMeta) {
 							!!writeMeta &&
 							options.css !== 'inline' &&
 							options.output.replace(
-								/(\.(umd|cjs|es|m))?\.(mjs|[tj]sx?)$/,
+								/(\.(umd|cjs|es|m))?\.([cm]js|[tj]sx?)$/,
```
Comment (rschristian) [inline src/index.js:484]: This was the only regex that needed to change, the others were functionally fine, but I updated them to match.

**Comment by MinJieLiu** [comment]
When can this pr be merged

File: src/utils.js [hunk 2]
```diff
@@ -43,3 +43,5 @@ export function safeVariableName(name) {
 	const identifier = normalized.replace(INVALID_ES3_IDENT, '');
 	return camelCase(identifier);
 }
+
+export const EXTENSION = /(\.(umd|cjs|es|m))?\.([cm]?[tj]sx?)$/;
```
Comment (rschristian) [inline src/utils.js:47]: Names are hard... 😅

Certainly open to suggestions

File: src/index.js [hunk 3]
```diff
@@ -482,10 +484,7 @@ function createConfig(options, entry, format, writeMeta) {
 						extract:
 							!!writeMeta &&
 							options.css !== 'inline' &&
-							options.output.replace(
-								/(\.(umd|cjs|es|m))?\.(mjs|[tj]sx?)$/,
-								'.css',
-							),
+							options.output.replace(EXTENSION, '.css'),
```
Comment (rschristian) [inline src/index.js:487]: This is the only regex that actually needed to be corrected. #802 missed this when adding in support, which is totally understandable given that the regex existed in 3 separate locations. Hopefully extracting this out makes it easier to handle.

File: src/utils.js (hunk 2 above)
Comment (rschristian) [reply src/utils.js:47]: Also (horrifyingly) supports the [`.cts` and `.mts` extensions](https://www.typescriptlang.org/docs/handbook/esm-node.html), as it looks like TS might head down that path

**Pull request merged**